notifier/tasks.py

```diff
@@ -122,7 +128,26 @@ def batch_digest_subscribers():
 
     from_dt, to_dt = _time_slice(settings.FORUM_DIGEST_TASK_INTERVAL)
 
-    logger.info("Beginning forums digest task: from_dt=%s to_dt=%s", from_dt, to_dt)
+    # Remove old tasks from the database so that the table doesn't keep growing forever.
+    ForumDigestTask.prune_old_tasks(settings.FORUM_DIGEST_TASK_GC_DAYS)
+
+    if self.request.retries == 0:
+        task, created = ForumDigestTask.objects.get_or_create(
+            from_dt=from_dt,
+            to_dt=to_dt,
+            defaults={'node': platform.node()}
+        )
+        if created:
+            logger.info("Beginning forums digest task: from_dt=%s to_dt=%s", from_dt, to_dt)
+        else:
```

Comment:
All right, excellent. I wasn't sure if there was a dependency on the current time on the system clock or something like that.